### PR TITLE
Jigsaw Game added

### LINKED ISSUE #4395: [New game]: Jigsaw Game
### 🎮 Game Request

Game logic and basic description:
Jigsaw Puzzle Game Description
A Jigsaw Puzzle game where players drag and drop pieces to fit them together and complete a picture.

Game Logic
Shuffle Pieces: Randomly place jigsaw pieces around the canvas.
Drag and Drop: Allow pieces to be draggable and snapped into place when close to their correct position.
Completion Check: Continuously check if all pieces are correctly placed to display a completion message.

### Point down the features

Jigsaw Puzzle Game Features
Piece Shuffling: Randomly distribute pieces on the canvas.
Drag and Drop: Enable dragging and snapping of pieces.
Completion Check: Detect when all pieces are correctly placed.
Completion Message: Display a message upon completing the puzzle.
Image Selection: Allow users to select different images for the puzzle.

### Select program in which you are contributing

GSSoC24

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

### PR DESCRIPTION
**Pull request by Vipullakum007**
## PR Description 📜

i have added Jigsaw game

Fixes #4395 

<hr>
 
## Mark the task you have completed ✅

- [x] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) & [CODE OF CONDUCT](https://github.com/kunjgit/GameZone/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I have followed proper naming convention showed in [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md)
- [x] I have added screenshot for website preview in assets/images 
- [x] I have added entries for my game in main README.md
- [x] I have added README.md in my folder 
- [ ] I have added working video of the game in README.md (optional)
- [x] I have specified the respective issue number for which I have requested the new game.

<hr>

## Add your screenshots(Optional) 📸

jigsaw_game_1
jigsaw_game_2

--- 
<br>

## Thank you soo much for contributing to our repository 💗

### CODE REVIEW

**Comment by kunjgit** [comment]
Thank you @Vipullakum007 ,for creating the PR and contributing to our GameZone 💗 
 Review team will review the PR and will reach out to you soon! 😇 
 Make sure that you have marked all the tasks that you are done with ✅. 
 Thank you for your patience! 😀

**Comment by Vipullakum007** [comment]
hey @kunjgit @ishita-43 , ihave resolved conflics also now you can merge it.

**Comment by Vipullakum007** [comment]
hey @ishita-43 , please review it so that i can create new PR.

File: Games/Periodic_Table_Game/README.md
Comment (ishita-43) [inline Games/Periodic_Table_Game/README.md:1]: if this game is not the part of the jigsaw game remove this if it is the part then include this in your game folder

File: Games/Toast_Notification/README.md
Comment (ishita-43) [inline Games/Toast_Notification/README.md:1]: if this game is not the part of the jigsaw game remove this if it is the part then include this in your game folder

File: assets/js/gamesData.json [hunk 1]
```diff
@@ -2019,6 +2024,11 @@
   "gameUrl": "path_finder",
   "thumbnailUrl": "pathfinder.png"
 },
+"407":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2027]: respectively remove these game entry

**Review by ishita-43 — CHANGES_REQUESTED** [review]
Add your game in readme.md file of the gamezone refer other entries for format 
make separate pr for the different games and include only game in one pr  that will help you in gaining more points for those pr
if you want to keep this in one only then tell me.

**Comment by kunjgit** [comment]
Thank you  @Vipullakum007 , for your valuable time and contribution in our GameZone 💗. 
 It’s our GameZone, so Let’s build this GameZone altogether !!🤝
  Hoping to see you soon with another PR again 😇 
 Wishing you all the best for your journey into Open Source🚀

**Comment by Vipullakum007** [comment]
hey @ishita-43 , all conflicts are changed and i have created new PR

**Pull request closed without merging**